### PR TITLE
Fix/doc code tests

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

The test to detect outdated code block in our docs had two issues that prevented us from parsing all code blocks. This PR makes the following changes:
* Removes mistune, as it is not compatible with admonitions and therefore prevented a correct parsing
* Implements custom code block parsing via regex
* Fixes the test adding all the files and code blocks that were not being properly checked

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

File: docs/demos/price_oracle_fsms.md [hunk 1]
```diff
@@ -28,7 +28,6 @@ The summary of the constituent FSMs is as follows:
 | FSM                     | States  | Start states  | Final states  | Events  | Non-trivial transitions (*)   |
 |-----------------------  |-------: |-------------: |-------------: |-------: |------------------------:  |
 | AgentRegistration       |      4  |            2  |            2  |      3  |                       3   |
-| SafeDeployment          |      5  |            1  |            1  |      8  |                      14   |
```
Comment (dvilelaf) [inline docs/demos/price_oracle_fsms.md:31]: The safe deployment was removed from the oracle

File: scripts/check_doc_links.py [hunk 2]
```diff
@@ -49,6 +49,8 @@
 HTTP_SKIPS = [
     "http://www.fipa.org/repository/ips.php3",
     "http://host.docker.internal:8545",
+    "http://host.docker.internal:8545",  # internal (ERR_NAME_NOT_RESOLVED)
+    "http://host.docker.internal:8545}",  # regex issue
```
Comment (jmoreira-valory) [inline scripts/check_doc_links.py:53]: Why repeated 3 times?

Comment (dvilelaf) [reply scripts/check_doc_links.py:53]: There are two internal docker links in our docs: the ones set as environment variable (written inside brackets) and the "normal ones". There's two problems with those 
* They are HTTP links
* The regex gets the } as part of the link in the ones inside brackets
So I needed to add both of them to both HTTP and HTTPS skips.

File: tests/test_docs/helper.py [hunk 3]
```diff
@@ -50,36 +46,23 @@ class CodeType(Enum):
...
-
-
-def type_filter(type_: Optional[str], b: Dict) -> bool:
-    """
-    Check Mistune code block is of a certain type.
-
-    If the field "info" is None, return False.
-    If type_ is None, this function always return true.
+def dedent_code(code: str, n_spaces: int) -> str:
+    """Removes n spaces at the beggining of each line"""
```
Comment (jmoreira-valory) [inline tests/test_docs/helper.py:50]: begining

File: tests/test_docs/helper.py [hunk 4]
```diff
@@ -50,36 +46,23 @@ class CodeType(Enum):
...
+    lines = [line[n_spaces:] for line in code.split("\n")]
+    return "\n".join(lines)
```
Comment (jmoreira-valory) [inline tests/test_docs/helper.py:52]: To be more precise, the code removes the firs 'n' characters of each line (not necessarily spaces)

**Review by jmoreira-valory — APPROVED** [review]

File: scripts/check_doc_links.py (hunk 2 above)
Comment (angrybayblade) [reply scripts/check_doc_links.py:53]: The URL on lines 51 and 52 are the same

File: scripts/check_doc_links.py [hunk 5]
```diff
@@ -41,14 +41,14 @@
 
 
 MAX_WORKERS = 10
-URL_REGEX = r'(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s)"]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s)"]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s)"]{2,}|www\.[a-zA-Z0-9]+\.[^\s)"]{2,})'
+URL_REGEX = r'(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s})"]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s})"]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s)"}]{2,}|www\.[a-zA-Z0-9]+\.[^\s)"}]{2,})'
```
Comment (dvilelaf) [inline scripts/check_doc_links.py:44]: I have now fixed this regex so it does not get the invalid char }

File: scripts/check_doc_links.py (hunk 2 above)
Comment (dvilelaf) [reply scripts/check_doc_links.py]: Fixed

**Review by jmoreira-valory — APPROVED** [review]
LGTM.
Hope we can integrate the mapping and skipped files in the .md as "magic comments" in a future improvement.

**Review by Adamantios — APPROVED** [review]

**Pull request merged**